app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityRegisterToActivityTest.java

```diff
@@ -0,0 +1,177 @@
+package com.sdp.movemeet.view.activity;
+
+import android.content.Intent;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.storage.StorageReference;
+import com.sdp.movemeet.backend.BackendManager;
+import com.sdp.movemeet.backend.firebase.firestore.FirestoreActivityManager;
+import com.sdp.movemeet.backend.providers.AuthenticationInstanceProvider;
+import com.sdp.movemeet.backend.serialization.ActivitySerializer;
+import com.sdp.movemeet.models.Activity;
+import com.sdp.movemeet.models.Sport;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.concurrent.CountDownLatch;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+@RunWith(AndroidJUnit4.class)
+public class ActivityDescriptionActivityRegisterToActivityTest {
```

Comment:
What are you testing in this class ?

Reply:
In this test, I unfortunately don't mock anything, but I am testing the fact that the user is properly registered in an activity. More precisely, this test goes through the process of adding the `userId` String to the `participantId` array field of the activity document in Firebase Firestore. If the test passes, it means that the user was properly registered and added to the list of participants!